src/Common/src/System/Net/SafeCloseSocket.Windows.cs

```diff
@@ -33,52 +33,56 @@ public ThreadPoolBoundHandle IOCPBoundHandle
         // Binds the Socket Win32 Handle to the ThreadPool's CompletionPort.
         public ThreadPoolBoundHandle GetOrAllocateThreadPoolBoundHandle(bool trySkipCompletionPortOnSuccess)
         {
+            // Check to see if the socket native _handle is already
+            // bound to the ThreadPool's completion port.
+            if (_released || _iocpBoundHandle == null)
```

Comment:
Makes sense.

Another idea: is there any harm in just returning the existing iocp-bound handle if _released is true and _iocpBoundHandle is not null?

Don't know if any of this matters in practice, so don't block on this...